### PR TITLE
Add automation for word docs, analgous to notebooks

### PR DESCRIPTION
**Pull request by hamelsmu**
I promised that I would also do a Word converter in our email discussion.  So, I'm opening this PR to fulfill my promises.  

## Summary of changes made:

- Rearranged `_action_files` such that word and notebooks both share the same naming function, and moved shared code into `_action_files/fast_template.py`
- Added Action Workflow that converts word to md
- Added scaffolding to `_config.yml` to guide people to specify their `url`, which is useful when someone hosts their blog as not root (for example a project site, or custom domain).  This value being null does not break anything.
- Added a `_notebooks/README.md` file, in a similar spirit that was done for notebooks. 

## Key Differences from notebook converter

- I do NOT have the requirement of having the magic keyword `/sync` in commit messages for word.  I thought that is too technical and wanted to keep it simple, assuming many people would just use the sync feature of GitHub Desktop.

- I'm injecting `{{site.url}}` into image links, and saving media under `/assets/img`.  I did this so people do not have to fiddle with moving images around or renaming links.

Please let me know if this interface is suitable for users, happy to iterate it on it as you suggest 🙇

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@
 title: Edit _config.yml to set your title!
 description: This is where the description of your site will go. You should change it by editing the _config.yml file. It can be as long as you like! Happy blogging... ❤
 github_username: jph00
+url: # the full url for your site for your site, e.g. http://username.github.io/blog
```
Comment (jph00) [inline _config.yml]: This is optional, at least in some cases, right? If so, move it after email and twitter, and add a comment above saying it's optional, and explain when it's needed.

Might it be possible to use the metadata from this plugin instead?: https://github.com/jekyll/github-metadata . I've used it for the social icons already.

**Review by jph00 — CHANGES_REQUESTED** [review]
This is looking great!

**Comment by hamelsmu** [comment]
@jph00 Great idea.  I made the following changes:

- Moved the scaffolding below email / twitter ( it is completely commented out, doesn't work well when it is null ) 
- Added the plugin to the `Gemfile` & `_config.yaml`
- Updated scripts to use `{{ site.url }}{{ site.baseurl }}` for image links, instead of just `{{ site.url }}`to reflect the new plugin 🎉 
- Updated the README to mention setting the parameters `url` and `baseurl` ONLY in the case of a custom domain
- Now users don't have to set anything in `_config.yaml` when using a project (non-root) pages, b/c of the cool plugin 🚀 

I tested it to make sure it works, works great!

**Comment by jph00** [comment]
Great!

**Pull request merged**